librosa/sequence.py

```diff
@@ -0,0 +1,696 @@
+    # if it's a float, make it a vector
+    prob = np.asarray(prob, dtype=np.float)
+
+    if prob.ndim == 0:
+        prob = np.tile(prob, n_states)
+
+    if prob.shape != (n_states,):
+        raise ParameterError('prob={} must have length equal to n_states={}'.format(prob, n_states))
+
+    if np.any(prob < 0) or np.any(prob > 1):
+        raise ParameterError('prob={} must have values in the range [0, 1]'.format(prob))
+
+    for i, prob_i in enumerate(prob):
+        transition[i] = (1. - prob_i) / (n_states - 1)
+        transition[i, i] = prob_i
+
+    return transition
+
+
+def transition_cycle(n_states, prob):
+    '''Construct a cyclic transition matrix over `n_states`.
+
+    The transition matrix will have the following properties:
+
+        - `transition[i, i] = p`
+        - `transition[i, i + 1] = (1 - p)`
+
+    This type of transition matrix is appropriate for state spaces
+    with cyclical structure, such as metrical position within a bar.
+    For example, a song in 4/4 time has state transitions of the form
+
+        1->{1, 2}, 2->{2, 3}, 3->{3, 4}, 4->{4, 1}.
+
+    Parameters
+    ----------
+    n_states : int > 1
+        The number of states
+
+    prob : float in [0, 1] or iterable, length=n_states
+        If a scalar, this is the probability of a self-transition.
+
+        If a vector of length `n_states`, `p[i]` is the probability of state
+        `i`'s self-transition.
+
+    Returns
+    -------
+    transition : np.ndarray, shape=(n_states, n_states)
+        The transition matrix
+
+    Examples
+    --------
+    >>> librosa.sequence.transition_cycle(4, 0.9)
+    array([[0.9, 0.1, 0. , 0. ],
+           [0. , 0.9, 0.1, 0. ],
+           [0. , 0. , 0.9, 0.1],
+           [0.1, 0. , 0. , 0.9]])
+    '''
+
+    if not isinstance(n_states, int) or n_states <= 1:
+        raise ParameterError('n_states={} must be a positive integer > 1')
+
+    transition = np.zeros((n_states, n_states), dtype=np.float)
+
+    # if it's a float, make it a vector
+    prob = np.asarray(prob, dtype=np.float)
+
+    if prob.ndim == 0:
+        prob = np.tile(prob, n_states)
+
+    if prob.shape != (n_states,):
+        raise ParameterError('prob={} must have length equal to n_states={}'.format(prob, n_states))
+
+    if np.any(prob < 0) or np.any(prob > 1):
+        raise ParameterError('prob={} must have values in the range [0, 1]'.format(prob))
+
+    for i, prob_i in enumerate(prob):
+        transition[i, np.mod(i + 1, n_states)] = 1. - prob_i
+        transition[i, i] = prob_i
+
+    return transition
+
+
+def transition_local(n_states, width, window='triangle', wrap=False):
+    '''Construct a localized transition matrix.
+
+    The transition matrix will have the following properties:
+
+        - `transition[i, j] = 0` if `|i - j| > width`
+        - `transition[i, i]` is maximal
+        - `transition[i, i - width//2 : i + width//2]` has shape `window`
+
+    This type of transition matrix is appropriate for state spaces
```

Comment:
Can we have such a sentence for `viterbi()` and `viterbi_disrciminative()` just mentioning an MIR problem saying what are the observations (or states) and briefly explain what the function will then calculate.